docs/configuration/index.md

```diff
@@ -1135,6 +1135,7 @@ The following configs only apply if the Overlord is running in remote mode. For
 |`druid.indexer.runner.taskAssignmentTimeout`|How long to wait after a task has been assigned to a Middle Manager before throwing an error.|`PT5M`|
 |`druid.indexer.runner.minWorkerVersion`|The minimum Middle Manager version to send tasks to. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.worker.version`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accommodate date comparisons. |"0"|
 | `druid.indexer.runner.parallelIndexTaskSlotRatio`| The ratio of task slots available for parallel indexing supervisor tasks per worker. The specified value must be in the range `[0, 1]`. |1|
+|`druid.indexer.runner.taskSlotLimits`| A map where each key is a task type, and the corresponding value represents the limit on the number of task slots that a task of that type can occupy on a worker. The key is a `String` that specifies the task type. The value can either be a Double or Integer. A `Double` in the range [0, 1], representing a ratio of the available task slots that tasks of this type can occupy. An `Integer` that is greater than or equal to 0, representing an absolute limit on the number of task slots that tasks of this type can occupy.|Empty map|
```

Comment:
Could you please provide an example as well ?

Reply:
How does this interact with compaction slots ?

Reply:
Example added! Good catch on the compaction slots. I'll need to test that, but based on the code, it looks like compaction slots availability will be checked twice (in the duty, and during the worker selection) if they're included in the `taskSlotLimits` map. If there's a conflict, some tasks might end up in a pending state for a while.

Reply:
Confirmed that the number of submitted tasks is linked to the compaction task slot limits. However, execution might be delayed if the custom limit is smaller than the one set for compaction.
![image](https://github.com/user-attachments/assets/71a7f10e-b23f-4282-80bd-f07a794a13ab)
For example here,  `taskSlotsMax = 3`, but in the overlord configuration, I have `druid.indexer.runner.taskSlotLimits={"compact": 2}`.

Reply:
I think the limit on compaction tasks (or kill tasks for that matter) should not be a concern.
This is a runtime property, typically controlled by an admin.
So, if an admin wants to restrict the number of concurrent compaction tasks, it is fair to honor that irrespective of the value of `compactionTaskSlotRatio` or `maxCompactionTaskSlots` set in the coordinator dynamic configs.

We just need to call it out clearly in the release notes and the docs of the new property.